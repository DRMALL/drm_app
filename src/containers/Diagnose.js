import React, { Component } from 'react'
import { View, Text, Image } from 'react-native'
import TabBarItem from '../components/units/TabBarItem'
import DiagnosisTab from '../components/units/DiagnosisTab'
import DiagnoseCategory from '../components/DiagnoseCategory'
import { diagnosisTabData } from '../utils/virtualData'

const diagnoseIconSelected = require('../images/tabbar_icons/tabbar_diagnosis_selected_x.png')
    , diagnoseIconNormal = require('../images/tabbar_icons/tabbar_diagnosis_normal.png')

export default class Diagnose extends Component {
  static navigationOptions = {
    tabBarIcon: ({ focused, tintColor }) => (
      <TabBarItem
        tintColor={tintColor}
        focused={focused}
        normalImage={diagnoseIconNormal} 
        selectedImage={diagnoseIconSelected} 
      />
    )
  };

  constructor(props) {
    super(props)
    this.state = (()=> {
      let diagTabState = {}
      diagnosisTabData.map((itemTabType, index)=> {
        if(index == 0) diagTabState[`tabTypeRow${index}`] = true
        else diagTabState[`tabTypeRow${index}`] = false
      })
      return diagTabState
    })()
  }

  pressTab(dt) {
    diagnosisTabData.map((itemTabType, index)=> {
      if(dt == index) this.setState({[`tabTypeRow${dt}`]: true})
      else this.setState({[`tabTypeRow${index}`]: false})
    })
  }

  render() {
    return(
      <View style={{height: '100%', paddingBottom: 45}}>
        <DiagnosisTab state={this.state} diagData={diagnosisTabData} pressTab={this.pressTab.bind(this)}/>
        <DiagnoseCategory {...this.props} />
      </View>
    )
  }
}