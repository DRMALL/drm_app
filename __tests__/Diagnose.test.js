import 'react-native'
import React from 'react'
import Diagnose from '../src/containers/Diagnose'
import renderer from 'react-test-renderer';

test('Diagnose test', () => {
  const tree = renderer.create(
    <Diagnose />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})