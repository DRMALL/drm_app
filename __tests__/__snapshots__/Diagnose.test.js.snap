// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Diagnose test 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "paddingBottom": 45,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "rgba(0,0,0,.06)",
        "borderWidth": 0.5,
      }
    }
  >
    <RCTScrollView
      horizontal={true}
      showsHorizontalScrollIndicator={false}
    >
      <View />
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#F7F7F7",
        "height": "100%",
      }
    }
  >
    <RCTScrollView
      refreshControl={
        <RefreshControlMock
          colors={
            Array [
              "#ff0000",
              "#00ff00",
              "#0000ff",
            ]
          }
          onRefresh={[Function]}
          progressBackgroundColor="#fff"
          refreshing={false}
          title="下拉刷新"
          titleColor="rgba(0,0,0,.6)"
        />
      }
    >
      <RCTRefreshControl />
      <View />
    </RCTScrollView>
  </View>
</View>
`;
